src/Aspire.Hosting.Azure.ServiceBus/AzureServiceBusExtensions.cs

```diff
@@ -124,19 +151,381 @@ public static IResourceBuilder<AzureServiceBusResource> AddQueue(this IResourceB
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    /// <example>
+    /// The following example creates an Azure Service Bus resource that runs locally is an emulator and referencing that
+    /// resource in a .NET project.
+    /// <code lang="csharp">
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// var serviceBus = builder.AddAzureServiceBus("myservicebus")
+    ///    .RunAsEmulator()
+    ///    .AddQueue("queue");
+    ///
+    /// builder.AddProject&lt;Projects.InventoryService&gt;()
+    ///        .WithReference(serviceBus);
+    ///
+    /// builder.Build().Run();
+    /// </code>
+    /// </example>
+    public static IResourceBuilder<AzureServiceBusResource> RunAsEmulator(this IResourceBuilder<AzureServiceBusResource> builder, Action<IResourceBuilder<AzureServiceBusEmulatorResource>>? configureContainer = null)
+    {
+        if (builder.ApplicationBuilder.ExecutionContext.IsPublishMode)
+        {
+            return builder;
+        }
+
+        // Add emulator container
+        var configHostFile = Path.Combine(Path.GetTempPath(), "AspireServiceBusEmulator", Path.GetRandomFileName() + ".json");
+
+        Directory.CreateDirectory(Path.GetDirectoryName(configHostFile)!);
+
+        var password = PasswordGenerator.Generate(16, true, true, true, true, 0, 0, 0, 0);
+
+        var customMountAnnotation = new ContainerMountAnnotation(
+                configHostFile,
+                AzureServiceBusEmulatorResource.EmulatorConfigJsonPath,
+                ContainerMountType.BindMount,
+                isReadOnly: false);
+
+        builder
+            .WithEndpoint(name: "emulator", targetPort: 5672)
+            .WithAnnotation(new ContainerImageAnnotation
+            {
+                Registry = ServiceBusEmulatorContainerImageTags.Registry,
+                Image = ServiceBusEmulatorContainerImageTags.Image,
+                Tag = ServiceBusEmulatorContainerImageTags.Tag
+            })
+            .WithAnnotation(customMountAnnotation);
+
+        var sqlEdgeResource = builder.ApplicationBuilder
+                .AddContainer($"{builder.Resource.Name}-sqledge",
+                    image: ServiceBusEmulatorContainerImageTags.AzureSqlEdgeImage,
+                    tag: ServiceBusEmulatorContainerImageTags.AzureSqlEdgeTag)
+                .WithImageRegistry(ServiceBusEmulatorContainerImageTags.AzureSqlEdgeRegistry)
+                .WithEndpoint(targetPort: 1433, name: "tcp")
+                .WithEnvironment("ACCEPT_EULA", "Y")
+                .WithEnvironment("MSSQL_SA_PASSWORD", password);
+
+        builder.WithAnnotation(new EnvironmentCallbackAnnotation((EnvironmentCallbackContext context) =>
+        {
+            var sqlEndpoint = sqlEdgeResource.Resource.GetEndpoint("tcp");
+
+            context.EnvironmentVariables.Add("ACCEPT_EULA", "Y");
```

Comment:
We should check with the ServiceBus team that it is OK to auto accept the EULA here.

cc @DamianEdwards